### PR TITLE
Allow specifying an optional printing process at the ennoblement step.

### PR DESCRIPTION
**Pull request by n1k0**
[User Story](https://www.notion.so/Teinture-ajout-de-2-proc-d-s-d-impression-4bb20cc0d05b400f9fd1d3848bea8cd1)

### CODE REVIEW

**Comment by paulboosz** [comment]
checks ok with gsheet ✅

File: public/data/processes.json [hunk 1]
```diff
@@ -7942,5 +7942,95 @@
...
+      "pma": 2.32444e-8,
+      "swe": 1.28288e-3,
+      "tre": 1.23719e-3,
+      "wtu": 0.0
+    },
+    "heat_MJ": 1.074e1,
+    "elec_pppm": 0.0,
+    "elec_MJ": 1.61,
+    "waste": 0.0,
+    "alias": null
```
Comment (magopian) [inline public/data/processes.json:8034]: Is this last process used somewhere? Is it relevant to the "pigment" or "substantive" printing process?

File: src/Views/Textile/Step.elm [hunk 2]
```diff
@@ -611,6 +647,26 @@ detailedView ({ inputs, funit, impact, daysOfWear, next, current } as config) =
...
+                        else if current.label == Label.Ennobling then
+                            Just ( "entrante", Step.getInputSurface inputs current )
+
+                        else
+                            Nothing
+                  in
+                  case surfaceInfo of
+                    Just ( dir, surface ) ->
+                        li [ class "list-group-item text-muted d-flex justify-content-center gap-2" ]
+                            [ span [] [ text <| "Surface étoffe (" ++ dir ++ ")\u{00A0}:" ]
```
Comment (magopian) [inline src/Views/Textile/Step.elm:664]: I don't remember, did we stick with "étoffe" or "tissu"?

File: tests/Server/RouteTest.elm [hunk 3]
```diff
@@ -274,10 +276,12 @@ textileEndpoints db =
                     , ( "countryMaking", "Code pays invalide: notACountryCode." )
                     , ( "mass", "La masse doit être supérieure ou égale à zéro." )
                     , ( "materials", "Format de matière invalide : notAnID." )
+                    , ( "surfaceMass", "Le grammage (surfaceMass) doit être compris entre 30 et 500 gr/m²." )
                     , ( "product", "Produit non trouvé id=notAProductID." )
                     , ( "disabledSteps", "Impossible d'interpréter la liste des étapes désactivées; Code étape inconnu: invalid" )
                     , ( "disabledFading", "La valeur ne peut être que true ou false." )
                     , ( "dyeingMedium", "Type de support de teinture inconnu: yolo" )
+                    , ( "printing", "Type d'impression inconnu: yolo" )
```
Comment (magopian) [inline tests/Server/RouteTest.elm:284]: Pretty sure this should be `yopo`: you only print once. But who am I to decide 🤣

**Review by magopian — APPROVED** [review]
Well done, that's one big piece of work 👏

File: public/data/processes.json (hunk 1 above)
Comment (n1k0) [reply public/data/processes.json:8034]: Nope, it's for the next card, do you want me to remove it?

Comment (n1k0) [reply public/data/processes.json:8034]: I'll keep it so it's already there.

Comment (magopian) [reply public/data/processes.json:8034]: As long as it's here on purpose, and not a bad copy/paste, I don't care ;)

**Pull request merged**